java-backend/src/main/java/org/kframework/backend/java/symbolic/InitializeRewriter.java

```diff
@@ -371,6 +373,21 @@ public ProcessProofRules invoke(GlobalContext rewritingContext, long initCounter
                         termContext.global());
             }
 
+            public ConjunctiveFormula getEvaluatedConstraint(org.kframework.backend.java.kil.Rule rule) {
+                termContext.setTopConstraint(null);
+                //We need this ConsTerm only to evaluate the constraint. That's why we use an empty first argument.
+                ConstrainedTerm constraintHolder = new ConstrainedTerm(
+                        ConjunctiveFormula.of(termContext.global()),
+                        rule.getRequires(),
+                        termContext).expandPatterns(true);
+
+                ConjunctiveFormula constraint = constraintHolder.constraint();
+                termContext.setTopConstraint(constraint);
+                //simplify the constraint in its own context, to force full evaluation of terms.
+                constraint = constraint.simplify(termContext);
```

Comment:
just to make sure, `constraint.simplify` ensures that each conjunct is simplified assuming all other conjuncts but itself, right?

Reply:
ok. i just double-checked that the simplify function handles that.